src/bin/psql/tab-complete.c

```diff
@@ -840,6 +847,9 @@ initialize_readline(void)
 }
 
 
+/* Max number of previous words scanned for completion context. */
+enum { LOOKBACK_LIMIT = 20 };
```

Comment:
How this number has been calculated?

Reply:
According to [CREATE TABLE definition](https://techdocs.broadcom.com/us/en/vmware-tanzu/data-solutions/tanzu-greenplum/6/greenplum-database/ref_guide-sql_commands-CREATE_TABLE.html) maximum number of keywords is **24**:
```sql
1 CREATE
2 GLOBAL
3 TEMP
4 TABLE
5 IF
6 NOT
7 EXISTS
8 TableName
9 (columns)
10 INHERITS
11 (...)
12 WITH
13 (...)
14 ON
15 COMMIT
16 PRESERVE
17 ROWS
18 TABLECPASE
19 <tablespace_name>
20 DISTRIBUTED
21 BY
22 (column)
23 PARTITION
24 BY
```
So, with a margin we can take **30 elements**.